Fixes for for hosting startup topic (#9478)
aspnet/Docs: aspnetcore/fundamentals/host/platform-specific-configuration.md

The hosting startup implementation is placed in the [runtime store](/dotnet/core/deploying/runtime-store). A compile-time reference to the assembly isn't required by the enhanced app.

After the hosting startup is built, a runtime store is generated using manifest project file using the [dotnet store](/dotnet/core/tools/dotnet-store) command.

```console
dotnet store --manifest <MANIFEST_FILE> --runtime <RUNTIME_IDENTIFIER> -o <OUTPUT_LOCATION>
```

> For runtime to be able to discover runtime store its location needs to be added to `DOTNET_SHARED_STORE` environment variable.

Reviewer: ```suggestion
For runtime to discover the runtime store, the runtime store's location is added to the `DOTNET_SHARED_STORE` environment variable.
```

**Modify and place the hosting startup's dependencies file**

> To be able to activate the enhancement without having package reference to it one should use `additionalDeps` feature of the runtime. It allows to extend applications library graph by providing set of additional `.deps.json` files that would get merged with apps own `.deps.json` on startup and would make hosting startup assembly discoverable and loadable.

Reviewer: ```suggestion
To activate the enhancement without having a package reference to the enhancement, use the `additionalDeps` feature of the runtime. `additionalDeps` allows you to:

* Extend the app's library graph by providing a set of additional *\*.deps.json* files to merge with the app's own *\*.deps.json* file on startup.
* Make the hosting startup assembly discoverable and loadable.
```

Note that I try to say consistently "compile-time reference." IIRC, I did that because it applies to both packages and non-packages (class libs). Can "having a package reference" say "having a compile-time reference," or is it better here the way that you have it, even considering my concern about non-package references?

> Easy way of generating additional deps file is running `dotnet publish` on runtime store manifest file refernced in previous section and then removing manifest reference from libraries and runtime section of resulting deps.json file.

Reviewer: ```suggestion
The recommended approach for generating the additional deps file is to:

 1. Execute `dotnet publish` on the runtime store manifest file referenced in the previous section.
 1. Remove the manifest reference from libraries and the `runtime` section of the resulting *\*deps.json* file.
```

> ``` diff

Reviewer: diff? ... This should be (???) ...

```
```json
```

Author: I was trying to show what is removed from .deps.json file.

Author: Sounds good.

{
  "runtimeTarget": {
    "name": ".NETCoreApp,Version=v2.1",
    "signature": "4ea77c7b75ad1895ae1ea65e6ba2399010514f99"
  },
  "compilationOptions": {},
  "targets": {
    ".NETCoreApp,Version=v2.1": {
-      "store.manifest/1.0.0": {
-        "dependencies": {
-          "StartupDiagnostics": "1.0.0"
-        },
-        "runtime": {
-          "store.manifest.dll": {}
-        }
-      },
      "StartupDiagnostics/1.0.0": {
        "runtime": {
          "lib/netcoreapp2.1/StartupDiagnostics.dll": {
            "assemblyVersion": "1.0.0.0",
            "fileVersion": "1.0.0.0"
          }
        }
      }
    }
  },
  "libraries": {
-    "store.manifest/1.0.0": {
-      "type": "project",
-      "serviceable": false,
-      "sha512": ""
-    },
    "StartupDiagnostics/1.0.0": {
      "type": "package",
      "serviceable": true,
      "sha512": "sha512-oiQr60vBQW7+nBTmgKLSldj06WNLRTdhOZpAdEbCuapoZ+M2DJH2uQbRLvFT8EGAAv4TAKzNtcztpx5YOgBXQQ==",
      "path": "startupdiagnostics/1.0.0",
      "hashPath": "startupdiagnostics.1.0.0.nupkg.sha512"
    }
  }
}
```

> Deps.json file is then placed into the following directory structure:

Reviewer: ```suggestion
Place the *\*.deps.json* file into the following location:
```

```
<ADDITIONAL_DEPS_PATH>/shared/<SHARED_FRAMEWORK_NAME>/<SHARED_FRAMEWORK_VERSION>/<ENHANCEMENT_ASSEMBLY_NAME>.deps.json
```

> Where `ADDITIONAL_DEPS_PATH` is location that is being added to `DOTNET_ADDITIONAL_DEPS`, `SHARED_FRAMEWORK_NAME` is shared framework required for this additionalDeps file, `SHARED_FRAMEWORK_VERSION` minimum shared framework version.

Reviewer: ```suggestion
* `ADDITIONAL_DEPS_PATH` &ndash; Location added to the `DOTNET_ADDITIONAL_DEPS` environment variable.
* `SHARED_FRAMEWORK_NAME` &ndash; Shared framework required for this additional dependencies file.
* `SHARED_FRAMEWORK_VERSION` &ndash; Minimum shared framework version.
```

> For runtime to be able to discover runtime store location  needs to be added to `DOTNET_ADDITIONAL_DEPS` environment variable.

Reviewer: I would make a grammar/style suggestion, but I can't grasp the meaning of the sentence. There's a prep phrase object missing or the subject of the sentence is missing.

Author: Hah. What I was trying to say is that you need to add `additionalDeps` location to  `DOTNET_ADDITIONAL_DEPS` environment variable.

> In the sample code (*RuntimeStore* project), building of runtime store and generating of additionalDeps is acomplished using [PowerShell](/powershell/scripting/powershell-scripting) script.

Reviewer: ```suggestion
In the sample app (*RuntimeStore* project), building the runtime store and generating the additional dependencies file is accomplished using a [PowerShell](/powershell/scripting/powershell-scripting) script.
```

For examples of how to set environment variables for various operating systems, see [Use multiple environments](xref:fundamentals/environments).

**Deployment**

To facilitate the deployment of a hosting startup in a multimachine environment, the sample app creates a *deployment* folder in published output that contains:

* The hosting startup runtime store.
* The hosting startup dependencies file.

> * A PowerShell script that creates or modifies the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES`, `DOTNET_SHARED_STORE` and `DOTNET_ADDITIONAL_DEPS` to support the activation of the hosting startup. Run the script from an administrative PowerShell command prompt on the deployment system.

Reviewer: Just missing a comma in the list ...

```suggestion
* A PowerShell script that creates or modifies the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES`, `DOTNET_SHARED_STORE`, and `DOTNET_ADDITIONAL_DEPS` to support the activation of the hosting startup. Run the script from an administrative PowerShell command prompt on the deployment system.
```
